src/features/domain-settings/hooks/useDomainSettingsForm.ts

```diff
@@ -0,0 +1,223 @@
+
+		return domainMetadata;
+	}, [domainId, provider]);
+
+	// Transaction handlers
+	const handleTransactionStart = (onLoadingHeader?: string) => {
+		setIsTransactionLoading(true);
+		setLoadingStatusText('Waiting approval from your wallet...');
+		onLoadingHeader && setFormHeader(onLoadingHeader);
+	};
+
+	const handleTransactionProcessing = (onLoadingText: string) => {
+		setLoadingStatusText(onLoadingText);
+	};
+
+	const handleTransactionSuccess = () => {
+		setIsTransactionLoading(false);
+		onStepUpdate(steps[2]);
+		onCheckMetadataLockStatus();
+		onGetDomainMetadata();
+	};
+
+	const handleTransactionError = (errorMessage: string) => {
+		setIsTransactionLoading(false);
+		setErrorText(errorMessage);
+		onStepUpdate(stepId === FormStep.COMPLETE ? steps[2] : steps[0]);
+	};
+
+	// Transactions
+	const onLockMetadataStatus = () => {
+		setErrorText(undefined);
+
+		const lockStatus = metadataLockedStatus;
+		const onLoadingText = `${
+			lockStatus ? 'Unlocking' : 'Locking'
+		} metadata... This may take up to 20 mins. Do not close this window or refresh your browser...`;
+
+		return executeTransaction(
+			sdk.lockDomainMetadata,
+			[domainId, !lockStatus, provider.getSigner()],
+			{
+				onStart: () => handleTransactionStart(),
+				onProcessing: () => handleTransactionProcessing(onLoadingText),
+				onSuccess: () => handleTransactionSuccess(),
+				onError: (error: any) => handleTransactionError(error.message),
+				invalidationKeys: [['user', { account, domainId, lockStatus }]],
+			},
+		);
+	};
+
+	const onSetAndLockMetadata = () => {
+		setErrorText(undefined);
+
+		const onLoadingHeader = 'Save & Lock Metadata';
+		const onLoadingText =
+			'Saving & locking metadata... \nThis may take up to 20 mins. Do not close this window or refresh your browser...';
+
+		return executeTransaction(
+			sdk.setAndLockDomainMetadata,
+			[domainId, details, provider.getSigner()],
+			{
+				onStart: () => handleTransactionStart(onLoadingHeader),
+				onProcessing: () => handleTransactionProcessing(onLoadingText),
+				onSuccess: () => handleTransactionSuccess(),
+
+				onError: (error: any) => handleTransactionError(error.message),
+				invalidationKeys: [['user', { account, domainId, details }]],
+			},
+		);
+	};
+
+	const onSetMetadata = () => {
+		setErrorText(undefined);
+
+		const onLoadingHeader = 'Save Metadata Changes';
+		const onLoadingText =
+			'Saving metadata changes... This may take up to 20 mins. \nDo not close this window or refresh your browser...';
+
+		return executeTransaction(
+			sdk.setDomainMetadata,
+			[domainId, details, provider.getSigner()],
+			{
+				onStart: () => handleTransactionStart(onLoadingHeader),
+				onProcessing: () => handleTransactionProcessing(onLoadingText),
+				onSuccess: () => handleTransactionSuccess(),
+				onError: (error: any) => handleTransactionError(error.message),
+				invalidationKeys: [['user', { account, domainId, details }]],
+			},
+		);
+	};
```

Comment:
Could create an enum for `Save`, `Save & Lock`, `Lock`, combine these functions, and take an enum value as parameter. Then you'd just need to set `onLoadingHeader` and `onLoadingText` based on enum value. This would reduce the size of this file by quite a bit!

Reply:
This file has expanded now with some of the changes, i.e. combining hooks! I've added a switch case to handle the loading text content as it was getting a bit messy within the transaction function itself. Let me know what you think of this one. Other files have reduced as a result of making these changes and only return one `onSubmitTransaction` rather than `onLockMetadata`, `onSetAndLockMetadata` and `onSetMetadata`.